siatune/core/rewriters/dump.py

```diff
@@ -45,9 +41,10 @@ def __call__(self, context: Dict) -> Dict:
         Returns:
             Dict: The context after rewriting.
         """
-        cfg = context.pop(self.key)
-        trial_id = session.get_trial_id()
-        tmp_path = self.get_temporary_path(f'{trial_id}.py')
-        setattr(context.get('args'), self.arg_name, tmp_path)
-        dump_cfg(cfg, tmp_path)
+        is_parsed = not isinstance(context['args'], list)
```

Comment:
How about specify the expected type as `argparse.Namespace`?
It would be better to understand what this line tries to check.